core/query/src/main/java/org/eclipse/rdf4j/query/QueryResults.java

```diff
@@ -458,7 +449,7 @@ private static boolean bindingSetsMatch(BindingSet bs1, BindingSet bs2, Map<BNod
 					IRI dt1 = leftLit.getDatatype();
 					IRI dt2 = rightLit.getDatatype();
 
-					if (dt1 != null && dt2 != null && dt1.equals(dt2)
+					if (dt1 != null && dt1.equals(dt2)
 							&& XMLDatatypeUtil.isValidValue(leftLit.getLabel(), dt1)
 							&& XMLDatatypeUtil.isValidValue(rightLit.getLabel(), dt2)) {
 						Integer compareResult = null;
```

Comment:
I was just wondering if this can't be turned into an int?

Reply:
Dunno. I just cleanup the redundant null check.